Prox/Prox/PlaceDetails/PlaceDetailsCardView.swift

```diff
@@ -186,19 +188,57 @@ class PlaceDetailsCardView: UIScrollView {
         if frame != newFrame { frame = newFrame }
     }
 
+    // TODO: don't duplicate PlaceCarousel code
+    func updateUI(forPlace place: Place) {
+        // Labels will gracefully collapse on nil.
+        titleLabel.text = place.name
+        categoryLabel.text = place.categories?.prefix(MaxDisplayedCategories).joined(separator: " • ")
+        urlLabel.text = place.url ?? nil
+
+        // TODO: error states for missing data.
+        updateHoursUI(place.hours)
+
+        updateReviewUI(fromProvider: place.yelpProvider, onView: yelpReviewView)
+        updateReviewUI(fromProvider: place.tripAdvisorProvider, onView: tripAdvisorReviewView)
+    }
+
+    private func updateReviewUI(fromProvider provider: ReviewProvider?, onView view: ReviewContainerView) {
+        guard let provider = provider else {
+            view.score = 0
+            view.numberOfReviewersLabel.text = "No data found" // TODO: error state.
+            return
+        }
+
+        // TODO: error states
+        view.score = provider.rating ?? 0
+        view.numberOfReviewersLabel.text = "\(provider.totalReviewCount ?? 0) Reviews"
+    }
+
+    private func updateHoursUI(_ hours: [DayOfWeek:OpenHours]?) {
+        let todayDayOfWeek = DayOfWeek.forDate(Date())
+
+        let primaryText: String
+        let secondaryText: String
+        if hours == nil {
+            primaryText = "Unknown"
+            secondaryText = "No hours data :("
```

Comment:
I put this in as a place holder until we had the specs for the error state and Ricardo's error states were text like this, just less... ridiculous. I separated the cases (no hours data vs. no for the current day) as a form of visual debugging.

Though I also see how trying to do the right thing the first time can be faster. In this case, hiding the view breaks the layout so I'm not sure it'd be worth the time to set up.

What do you think? When we're missing specs, is it better to do whatever is easiest or to try to get it right?